l3kernel/l3cctab.dtx

```diff
@@ -54,7 +54,8 @@
 % range. For other engines, only the $8$-bit range ($0$--$255$) is covered by
 % such tables. The implementation of category code tables in \pkg{expl3}
 % also saves and restores the \TeX{} \tn{endlinechar} primitive value, meaning
-% they could be used for example to implement \cs{ExplSyntaxOn}.
+% they could be used for example to implement
+% a command similar to \cs{ExplSyntaxOn}.
```

Comment:
I'm not sure I follow why you want to change this: as far as I can see, it would work (though we've re-thought this area).

Reply:
@jlaurens  `\obeylines` is affected as the `\endlinechar`  is reset not because of the number of catcodes changed.

Reply:
@jlaurens The number of catcodes reset is not documented for `\ExplSyntax...` and arguable should be 'the lot' - it's a tradeoff with performance

Reply:
@davidcarlisle 
Exactly. Whereas, `\ExplSyntax...` only changes the value of  `\endlinechar`, selecting  a cctab will change both the value of  `\endlinechar *and* the category code of ^^M.

Reply:
That's an implementation detail: ExplSyntax could set the catcode of ^^M it doesn't matter much as ^^M will hardly ever appear in a document anyway since endlinechar is -1 it is not added at ends of lines

Reply:
Things are more clear to me: next commit includes
- change in cctab removed
- more reference to the "current category code regime"
- `\endlinechar` added
- nesting not supported
- precaution while changing category code due to performance reasons